fxa-rust-client/src/oauth.rs

```diff
@@ -0,0 +1,333 @@
+use errors::*;
+use regex::Regex;
+use url::Url;
+
+// These character ranges are from the OAuth RFC,
+// https://tools.ietf.org/html/rfc6749#section-3.3
+const VALID_SCOPE_VALUE: &str = r"^[\x21\x23-\x5B\x5D-\x7E]+$";
+const VALID_SHORT_NAME_VALUE: &str = r"^[a-zA-Z0-9_]+$";
+const VALID_FRAGMENT_VALUE: &str = r"^[a-zA-Z0-9_]+$";
```

Comment:
nit: for clarify, I think we should add an additional comment along the lines of "// These are additional restrictions that FxA imposes" just above the `VALID_SHORT_NAME_VALUE` regex.  Otherwise, it might appear to the reader that RFC6749 defines all three of these regexes, which is not the case.

(And I should probably take my own advice over the linked RP as well)